core/art/views.py

```diff
@@ -1206,56 +1252,61 @@ def registerARTTest(request):
         _logger.warning(data['message'] + str(request.session['requestParams']))
         return HttpResponse(json.dumps(data), status=422, content_type='application/json')
 
-    # Checking if provided pandaid exists in panda db
-    query = {'pandaid': pandaid}
-    values = ('pandaid', 'jeditaskid', 'username', 'computingsite', 'jobname')
-    jobs = []
-    jobs.extend(CombinedWaitActDefArch4.objects.filter(**query).values(*values))
-    try:
-       job = jobs[0]
-    except:
-        data = {'exit_code': -1, 'message': "Provided pandaid does not exists"}
-        _logger.warning(data['message'] + str(request.session['requestParams']))
-        return HttpResponse(json.dumps(data), status=422, content_type='application/json')
-
-    # Checking whether provided pandaid is art job
-    if 'username' in job and job['username'] != 'artprod':
-        data = {'exit_code': -1, 'message': "Provided pandaid is not art job"}
-        _logger.warning(data['message'] + str(request.session['requestParams']))
-        return HttpResponse(json.dumps(data), status=422, content_type='application/json')
-
-    # Preparing params to register art job
-    branch = concat_branch({'nightly_release_short':nightly_release_short, 'project': project, 'platform': platform})
-    if 'computingsite' in job:
-        computingsite = job['computingsite']
-    if 'jeditaskid' in job:
-        jeditaskid = job['jeditaskid']
-
-        # get files -> extract log tarball name, attempts
-        files = []
-        fquery = {'jeditaskid': jeditaskid, 'pandaid': pandaid, 'type__in': ('pseudo_input', 'input', 'log')}
-        files.extend(Filestable4.objects.filter(**fquery).values('jeditaskid', 'pandaid', 'fileid', 'lfn', 'type', 'attemptnr'))
-        # count of attempts starts from 0, for readability change it to start from 1
-        if len(files) > 0:
-            input_files = [f for f in files if f['type'] in ('pseudo_input', 'input')]
-            if len(input_files) > 0:
-                attemptnr = 1 + max([f['attemptnr'] for f in input_files])
-                inputfileid = max([f['fileid'] for f in input_files])
-            log_lfn = [f['lfn'] for f in files if f['type'] == 'log']
-            if len(log_lfn) > 0:
-                try:
-                    tarindex = int(re.search('.([0-9]{6}).log.', log_lfn[0]).group(1))
-                except:
-                    _logger.info('Failed to extract tarindex from log lfn')
-                    tarindex = None
-    if 'jobname' in job:
+    # Only check ART Grid tests
+    if test_type and test_type == 'local':
+        branch = concat_branch({'nightly_release_short':nightly_release_short, 'project': project, 'platform': platform})
+        attemptnr = 1
+    else:
```

Comment:
and here it can be simple if test_type == "grid", instead of if-else, after the previous two lines moved to their places

Reply:
Since you have added test_type == "grid" to the existing jobs in the ART tables, this has been changed as suggested.